src/components/application_manager/src/commands/mobile/perform_interaction_request.cc

```diff
@@ -425,17 +424,21 @@ void PerformInteractionRequest::ProcessPerformInteractionResponse(
     } else if (msg_params.keyExists(strings::manual_text_entry)) {
       msg_params[strings::trigger_source] =
           mobile_apis::TriggerSource::TS_KEYBOARD;
+      if (msg_params[strings::manual_text_entry].empty()) {
+        msg_params.erase(strings::manual_text_entry);
+      }
     }
   }
 
-  DisablePerformInteraction();
-
-  const char* return_info = (info.empty()) ? NULL : info.c_str();
   const SmartObject* response_params =
       msg_params.empty()
       ? NULL
       : &msg_params;
-  SendResponse(result, result_code, return_info, response_params);
+
+  if (mobile_apis::InteractionMode::BOTH != interaction_mode_) {
+    DisablePerformInteraction();
+    SendResponse(ui_result_, ui_resultCode_, info.c_str(), response_params);
```

Comment:
Please clarify, we need waiting for response from VR and UI if SDL receives interaction mode VR and MANUAL_ONLY.